packages/wdio-browserstack-service/src/util.ts

```diff
@@ -67,3 +85,439 @@ export function getParentSuiteName(fullTitle: string, testSuiteTitle: string): s
+    return world.pickle.uri + '_' + world.pickle.astNodeIds.join(',')
+}
+
+export function getCloudProvider(browser: Browser<'async'> | MultiRemoteBrowser<'async'>): string {
+    if (browser.options && browser.options.hostname && browser.options.hostname.includes('browserstack')) {
+        return 'browserstack'
+    }
+    return 'unknown_grid'
+}
+
+export function isBrowserstackSession(browser: Browser<'async'> | MultiRemoteBrowser<'async'>): boolean {
+    return getCloudProvider(browser).toLowerCase() == 'browserstack'
+}
+
+export function getScenarioExamples(world: ITestCaseHookParameter) {
+    const scenario = world.pickle
+
+    // no examples present
+    if ((scenario.astNodeIds && scenario.astNodeIds.length <= 1) || scenario.astNodeIds == undefined) return
+
+    const pickleId: string = scenario.astNodeIds[0]
+    const examplesId: string = scenario.astNodeIds[1]
+    const gherkinDocumentChildren = world.gherkinDocument.feature?.children
+
+    let examples: string[] = []
+
+    gherkinDocumentChildren?.forEach(child => {
+        if (child.rule) {
+            // handle if rule is present
+            child.rule.children.forEach(childLevel2 => {
+                if (childLevel2.scenario && childLevel2.scenario.id == pickleId && childLevel2.scenario.examples) {
+                    const passedExamples = childLevel2.scenario.examples.flatMap((val) => (val.tableBody)).find((item) => item.id == examplesId)?.cells.map((val) => (val.value))
+                    if (passedExamples) examples = passedExamples
+                }
+            })
+        } else if (child.scenario && child.scenario.id == pickleId && child.scenario.examples) {
+            // handle if scenario outside rule
+            const passedExamples = child.scenario.examples.flatMap((val) => (val.tableBody)).find((item) => item.id == examplesId)?.cells.map((val) => (val.value))
+            if (passedExamples) examples = passedExamples
+        }
+    })
+
+    if (examples.length) return examples
+    return
+}
+
+export function removeAnsiColors(message: string): string {
+    // https://stackoverflow.com/a/29497680
+    // eslint-disable-next-line no-control-regex
+    return message.replace(/[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g, '')
+}
+
+export function getLogTag(eventType: string): string {
+    if (eventType == 'TestRunStarted' || eventType == 'TestRunFinished') {
+        return 'Test_Upload'
+    } else if (eventType == 'HookRunStarted' || eventType == 'HookRunFinished') {
+        return 'Hook_Upload'
+    } else if (eventType == 'ScreenshotCreated') {
+        return 'Screenshot_Upload'
+    } else if (eventType == 'LogCreated') {
+        return 'Log_Upload'
+    }
+    return 'undefined'
+}
+
+function keepAliveAgent () {
+    return {
+        http: new http.Agent({ keepAlive: true }),
+        https: new https.Agent({ keepAlive: true }),
+    }
+}
```

Comment:
Instead of having this function which doesn't do much, define a default request config object, e.g. 
```
const DEFAULT_REQUEST_CONFIG = {
    agent: {
        http: new http.Agent({ keepAlive: true }),
        https: new https.Agent({ keepAlive: true }),
    },
    headers: {
        'Content-Type': 'application/json',
        'X-BSTACK-OBS': 'true'
    },
}
```
and assign it as parameter like: